packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts

```diff
@@ -0,0 +1,25 @@
+import AWS from 'aws-sdk';
+import aws from './aws';
+import { loadConfiguration } from '../configuration-manager';
+import { $TSContext } from 'amplify-cli-core';
+
+export class STS {
+  public sts: AWS.STS;
+
+  constructor(private readonly context: $TSContext, options = {}) {
+    const instancePromise = (async () => {
```

Comment:
we need to get away from this pattern of async constructors -- it's really bad practice. Instead follow the getInstance pattern we use for the s3 and amplifyBackend clients

Reply:
Ok

Reply:
So do you want me to change this or not?

Reply:
Ok

Reply:
So do you want me to change this or not?

Reply:
Ok

Reply:
So do you want me to change this or not?

Reply:
So do you want me to change this or not? Your comment is vague here

Reply:
So do you want me to change this or not? Your comment is vague here

Reply:
So do you want me to change this or not? Your comment is vague here